Runtimes/Core/cmake/modules/EmitSwiftInterface.cmake

```diff
@@ -9,63 +9,74 @@ function(emit_swift_interface target)
   # Generate the target-variant binary swift module when performing zippered
   # build
   if(SwiftCore_VARIANT_MODULE_TRIPLE)
-    set(variant_module_tmp_dir "${CMAKE_CURRENT_BINARY_DIR}/${target}-${SwiftCore_VARIANT_MODULE_TRIPLE}")
-    file(MAKE_DIRECTORY "${variant_module_tmp_dir}")
+    # Clean this up once CMake has nested swiftmodules in the build directory:
```

Comment:
This does make some complexity. We already have to have the custom handling here, why not consider doing this unconditionally and always generating the thick module layout?